nilearn/_utils/tests/test_data_gen.py

```diff
@@ -531,3 +571,119 @@ def test_generate_fake_fmri_error(rng):
             block_size=None,
             random_state=rng,
         )
+
+
+@pytest.mark.parametrize(
+    "shapes", [[(2, 3, 5, 7)], [(5, 5, 5, 3), (5, 5, 5, 5)]]
```

Comment:
Would it be possible to consider only two cases ? The point is to keep test time under control.

Reply:
Do you mean 2 cases per test or per parameter? Should other tests also reduce them to 2?

Reply:
per parameter, because the number of combinations quickly explodes.

Reply:
Oh, then this should be ok. There are 2 lists tested here. I reduced parameters for `generate_fake_fmri` though. Most of the cases are skipped when `n_block` is None anyway.